src/Reflection/EntityFieldsViaMagic.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace PHPStan\Reflection;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use PHPStan\Type\MixedType;
+
+/**
+ * Allows field access via magic methods
+ *
+ * See \Drupal\Core\Entity\ContentEntityBase::__get and ::__set.
+ */
+class EntityFieldsViaMagic implements PropertiesClassReflectionExtension {
+
+  public function hasProperty(ClassReflection $classReflection, string $propertyName): bool {
+    return $classReflection->isSubclassOf(ContentEntityBase::class) &&
```

Comment:
We should target the specific class which supports `__get` and `__set` to the `$values` and `$fields` properties. I believe this is `FieldableEntityInterface`.

Reply:
You're right, `FieldableEntityInterface` is the perfect fit. A lot of code I have seen incorrectly uses its parent interface, `EntityInterface`, though. Here is an example from Core:
````php
public function Drupal\Core\Entity\EntityListBuilder::buildRow(EntityInterface $entity);
````

Please decide if which interface we should ask for.